### ISSUE TITLE
add  state=latest to  FreeBSD_pkg

### ISSUE THREAD

**Issue by nusenu** [ansible/ansible#32327]
##### ISSUE TYPE
 - ~Bug  Report~
- Feature Idea

##### COMPONENT NAME
package module

##### ANSIBLE VERSION
```
v2.4.1.0-1
```
and
```
v2.4.2.0-0.1.beta1
```

##### OS / ENVIRONMENT

target: FreeBSD

##### STEPS TO REPRODUCE
Run the package module with 
```
state: latest
```
against a FreeBSD target.

example: but change `present` to `latest`:
https://github.com/nusenu/ansible-relayor/blob/master/tasks/main.yml#L83

##### EXPECTED RESULTS
package should be installed/updated.

##### ACTUAL RESULTS
```
failed: [...] "msg": "value of state must be one of: present,absent, got: latest"
```

If `latest` is not supported on FreeBSD and will not be supported, than the documentation must be updated.

**Comment by bcoca** [comment]
reclassifying as not all package managers are required to support latest

@nusenu it would help to know which package manager you are using on FreeBSD

**Comment by nusenu** [comment]
If the documentation says 'latest' is a valid value than this issue should not be considered a 'feature_idea'?

package manager: pkg

**Comment by bcoca** [comment]
@nusenu i would consider that a documentation bug, as package was made to only enforce present/absent #32397

**Comment by nusenu** [comment]
Brian Coca:
> @nusenu i would consider that a documentation bug, as package was made to only enforce present/absent

correct, and this issue is about fixing that bug in the documentation

maybe add "(not supported on ...)" next to "latest"?

**Comment by bcoca** [comment]
see PR linked above

**Comment by pilou-** [comment]
Documentation has been updated (`latest` has been removed, see PR linked above). @misc could you close this one ?

**Comment by bcoca** [comment]
@pilou- , not really, this is a 'feature idea' at this point

**Comment by antoinell** [comment]
The version of pkgng in the repository here https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/pkgng.py has the option state=latest. See https://github.com/ansible/ansible/pull/41623 .

**Comment by bcoca** [comment]
closed by #41623
